BrainPortal/app/controllers/custom_filters_controller.rb

```diff
@@ -58,19 +62,34 @@ def create #:nodoc:
     @custom_filter.data.merge! params[:data]
 
     @custom_filter.user_id = current_user.id
-
     @custom_filter.save
 
     if @custom_filter.errors.empty?
       flash[:notice] = "Filter successfully created."
+    else
+      flash[:error] = "Error creating filter."
+    end
+
+    respond_to do |format|
+      format.html { redirect_to :controller => :userfiles, :action => :index, :format => :html } if filter_param == UserfileCustomFilter.name
+      format.html { redirect_to :controller => :tasks, :action => :index, :format => :html } if filter_param == TaskCustomFilter.name
     end
 
   end
 
+  def show #:nodoc:
```

Comment:
the show custom filter page layout should be like the new custom filter page layout; they look different